framework/tests/viewpointtest/SuperConstructorCalls.java

```diff
@@ -0,0 +1,51 @@
+// Test case for EISOP issue #782:
+// https://github.com/eisop/checker-framework/issues/782
+import viewpointtest.quals.*;
+
+public class SuperConstructorCalls {
+
+    public SuperConstructorCalls() {}
+
+    @SuppressWarnings({"inconsistent.constructor.type", "super.invocation.invalid"})
+    public @ReceiverDependentQual SuperConstructorCalls(@ReceiverDependentQual Object str) {}
+
+    @SuppressWarnings({"inconsistent.constructor.type", "super.invocation.invalid"})
+    public @A SuperConstructorCalls(@A float number) {}
+
+    class Inner extends SuperConstructorCalls {
+        public Inner() {}
+
+        public Inner(Object obj) {
+            super();
+        }
+
+        public Inner(int number, @Top Object obj) {
+            super(obj);
+        }
+
+        // :: warning: (inconsistent.constructor.type)
+        public @A Inner(int number, @A Object obj1, @B Object obj2) {
+            super(obj1);
+        }
+
+        // :: warning: (inconsistent.constructor.type)
+        public @B Inner(int number, @B Object obj1, @A Object obj2, @Top Object obj3) {
```

Comment:
It would be good if you used the parameters more consistently - e.g. the first parameter is the one you actually use and the ones that are there only for overloading should make that clearer, e.g. by being named `dummy`.
This one is particularly confusing, as the meaning for `obj1` changes to the ones before.

Reply:
Thanks, I have updated the argument list and naming.